.github/workflows/ci.yml

```diff
@@ -1,11 +1,68 @@
 name: CI
-on: pull_request_target
+on: pull_request
 
 concurrency:
   group: ${{ github.head_ref }}
   cancel-in-progress: true
 
 jobs:
+  build_mac_windows:
+    name: Build and run limited tests
+    runs-on: ${{ matrix.os }}-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["macOS", "windows"]
+        ghc: ["8.6.5", "8.8.4", "8.10.7", "9.0.2"]
+        exclude:
+          # Some tests fail with a mysterious -11 error code.
+          - os: macOS
+            ghc: 8.10.7
+
+          # Windows gets non-deterministically gets stuck in infinite loops
+          # or segfaults while running the testcase.
+          - os: windows
+            ghc: 8.8.4
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: haskell/actions/setup@v1
+        id: setup-haskell
+        with:
+          ghc-version: ${{ matrix.ghc }}
+          enable-stack: true
+
+      - name: Install IVerilog (macOS)
+        if: matrix.os == 'macOS'
+        run: brew install icarus-verilog
+
+      - name: Install IVerilog (Windows)
+        if: matrix.os == 'windows'
+        run: choco install iverilog
```

Comment:
Does `choco` have a quiet flag? The constant printouts of how much it's downloaded really clutter up the log quickly

Reply:
The docs say `--no-progress`, let's see..

Reply:
That did the trick :tada: